File: go/chef.go
package jrmclient

import (
	"context"
)

// Chef submits Lisp source code (64 lines or fewer) to "The Chef" for a
// roast, via POST /api/v1/chef. Requires a bearer token (paid tier) and the
// caller's own Google Gemini API key, sent as the x-goog-api-key header.
// The response body is the roast as plain text.
func (c *Client) Chef(ctx context.Context, lispCode, geminiAPIKey string) (string, error) {
	var out string
	err := c.doRequestRaw(ctx, "POST", "/api/v1/chef", nil, []byte(lispCode), requestOptions{
		contentType: "text/plain",
		authorize:   true,
		extraHeader: map[string]string{"x-goog-api-key": geminiAPIKey},
	}, &out)
	if err != nil {
		return "", err
	}
	return out, nil
}

File: python/src/jrm_code_client/chef.py
"""POST /api/v1/chef"""

from __future__ import annotations

from .client import _BaseClient


class _ChefMixin(_BaseClient):
    def chef(self, lisp_code: str, gemini_api_key: str) -> str:
        """Submit LISP_CODE (64 lines or fewer) to "The Chef" for a roast,
        via POST /api/v1/chef. Requires ``client.token`` to be set (paid
        tier) and GEMINI_API_KEY, the caller's own Google Gemini API key,
        sent as the x-goog-api-key header. Returns the roast as plain text.
        """
        response = self._request(
            "POST",
            "/api/v1/chef",
            data=lisp_code.encode("utf-8"),
            content_type="text/plain",
            authorize=True,
            extra_headers={"x-goog-api-key": gemini_api_key},
        )
        return response.text
